### PR TITLE
Fix Switch's alignment when it's used with a left/right aligned FormFieldLabel

### LINKED ISSUE #3109: Create an example showing FormField and Switch
We have had requests to have a label to the left of the switch. Our current recommendation is to do this with a FormField. We should add a code example to our site (https://www.saltdesignsystem.com/salt/components/switch/examples).

### PR DESCRIPTION
**Pull request by libertymayc**
closes #3109

### CODE REVIEW

**Review by joshwooding — CHANGES_REQUESTED** [review]
I'm not a fan of these changes, we should find a way to do this without requiring users to add a class.

**Comment by libertymayc** [comment]
> I'm not a fan of these changes, we should find a way to do this without requiring users to add a class.

me too. are you thinking via a prop? If we use the form field label it will need styling since it's different from the Switch label. as mentioned in the description it also wouldn't have the interaction, plus makes the component itself technically have two labels since Switch is wrapped in one itself

**Comment by joshwooding** [comment]
> > I'm not a fan of these changes, we should find a way to do this without requiring users to add a class.
> 
> me too. are you thinking via a prop? If we use the form field label it will need styling since it's different from the Switch label. as mentioned in the description it also wouldn't have the interaction, plus it makes the component itself have two labels since Switch is wrapped in one itself 👎

I think the expectation is for it to look different anyway. It's more making sure Switch works with FormField

File: site/src/examples/switch/LeftAlignedLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { ReactElement } from "react";
+import { FormField, FormFieldLabel, Switch } from "@salt-ds/core";
+
+export const LeftAlignedLabel = (): ReactElement => (
+  <>
+    <FormField labelPlacement="left">
+      <FormFieldLabel>Active</FormFieldLabel>
+      <Switch />
+    </FormField>
+  </>
```
Comment (joshwooding) [inline site/src/examples/switch/LeftAlignedLabel.tsx:10]:
```suggestion
  <div>
    <FormField labelPlacement="left">
      <FormFieldLabel>Active</FormFieldLabel>
      <Switch />
    </FormField>
  </div>
```

File: site/src/examples/form-field/WithControls.tsx [hunk 2]
```diff
@@ -121,6 +121,10 @@ export const WithCheckboxAndRadioButton = (): ReactElement => {
           isCheckboxError ? "Must select at least one option. " : ""
         }`}</FormFieldHelperText>
       </FormField>
+      <FormField>
```
Comment (joshwooding) [inline site/src/examples/form-field/WithControls.tsx:124]: This example isn't loading?

Comment (libertymayc) [reply site/src/examples/form-field/WithControls.tsx:124]: fixed

File: packages/core/src/switch/Switch.css [hunk 3]
```diff
@@ -30,10 +30,6 @@
   box-sizing: border-box;
 }
 
-.saltSwitch-label {
```
Comment (origami-z) [inline packages/core/src/switch/Switch.css:33]: Need this for default switch label

**Review by bhoppers2008 — DISMISSED** [review]

File: packages/core/src/switch/Switch.css [hunk 4]
```diff
@@ -119,3 +119,10 @@
 .salt-density-high .saltSwitch-thumb {
   border: 0;
 }
+
+/* Styles applied when switch is inside a form field with label left or right, ensuring alignment with base / input height */
+.saltFormField-labelRight .saltSwitch,
+.saltFormField-labelLeft .saltSwitch {
```
[the remark this answers is not in the document]
Comment (origami-z) [reply packages/core/src/switch/Switch.css:125]: added new

**Review by joshwooding — APPROVED** [review]

File: .changeset/rare-turkeys-change.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/core": patch
+---
+
+Fix padding when Switch used with left/right aligned FormFieldLabel
```
Comment (joshwooding) [inline .changeset/rare-turkeys-change.md:5]:
```suggestion
Fixed padding when a Switch is used with a left/right aligned FormFieldLabel.
```

**Review by joshwooding — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Pull request merged**